docs/sync-overview.md

```diff
@@ -0,0 +1,57 @@
+
+[`SyncManager`](https://github.com/mozilla/application-services/blob/main/components/sync_manager/src/manager.rs) is responsible for performing the high-level parts of the sync process:
+  - The consumer code calls it's `sync()` function to start the sync, passing
+    in a [`SyncParams`](https://mozilla.github.io/application-services/rust-docs/sync_manager/msg_types/struct.SyncParams.html) object in, which describes what should be synced.
+  - `SyncManager` checks if we are currently in a backoff period and should wait before contacting the server again.
+  - `SyncManager` creates invididual `SyncEngine` instances for each component that should be synced.  We currently use 2 different methods for this:
+    - The older method is for the `SyncManager` to hold a weakref to a `Store` use that to create the `SyncEngine` (tabs and places).  The `SyncEngine` uses the `Store` for database access, see the [`TabsStore`](https://mozilla.github.io/application-services/rust-docs/tabs/struct.TabsStore.html) for an example.
+    - The newer method is for the components to provide a function to create the `SyncEngine`, hiding the details of how that engine gets created (autofill/logins).  These components also define a `Store` instance for the `SyncEngine` to use, but it's all transparent to the `SyncManager`.  (See [`autofill::get_registered_sync_engine()`](https://mozilla.github.io/application-services/rust-docs/autofill/db/store/fn.get_registered_sync_engine.html) and [`autofill::db::store::Store`](https://mozilla.github.io/application-services/rust-docs/autofill/db/store/struct.Store.html))
+  - For components that use local encryption, `SyncManager` passes the local encryption key to their `SyncEngine`
+  - Finally, calls `sync_multiple()` function from the `sync15` crate, sending it the `SyncEngine` instances.  `sync_multiple()` then calls the `sync()` function for each individual `SyncEngine`
+
+### Sync engines
+  - [`SyncEngine`](https://github.com/mozilla/application-services/blob/main/components/support/sync15-traits/src/engine.rs) is defined in the `support/sync15-traits` crate and defines the interface for syncing a component.
+  - A new `SyncEngine` instance is created for each sync
+  - `SyncEngine.apply_incoming()` does the main work.  It is responsible for processing incoming records from the server in order to update the local records and calculating which local records should be synced back.
+
+## The `apply_incoming` pattern
+
+`SyncEngine` instances are free to implement `apply_incoming()` any way they want, but the most components follow a general pattern.
+
+### Database Tables
+
+   - The local table stores records for the local application
+   - The mirror table stores the last known record from the server
+   - The staging temporary table stores the incoming records that we're currently processing
+   - Each table contains a `guid` as its primary key.  A record will share the same `guid` for the local/mirror/staging table.
```

Comment:
There's also a "metadata" table that stores the collection's guid and timestamp.